tests/setup_test.go

```diff
@@ -7215,3 +7215,242 @@ func TestProjectRunWebhookRedelivery(t *testing.T) {
 		}
 	})
 }
+
+func TestGetProjectCommitStatusDeliveries(t *testing.T) {
+	t.Parallel()
+
+	config := `
+		{
+			runs: [
+			{
+				name: 'run01',
+				tasks: [
+				{
+					name: 'task01',
+					runtime: {
+					containers: [
+						{
+						image: 'alpine/git',
+						},
+					],
+					},
+					steps: [
+					{ type: 'run', command: 'env' },
+					],
+				},
+				],
+			},
+			],
+		}
+	`
+
+	dir := t.TempDir()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sc := setup(ctx, t, dir, withGitea(true))
+	defer sc.stop()
+
+	giteaToken, tokenUser01 := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+	gwUser01Client := gwclient.NewClient(sc.config.Gateway.APIExposedURL, tokenUser01)
+	gwUserAdminClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, sc.config.Gateway.AdminToken)
+
+	_, _, err := gwUserAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: agolaUser02})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	user02Token, _, err := gwUserAdminClient.CreateUserToken(ctx, agolaUser02, &gwapitypes.CreateUserTokenRequest{TokenName: "token01"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	gwUser02Client := gwclient.NewClient(sc.config.Gateway.APIExposedURL, user02Token.Token)
+
+	giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+	giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	giteaRepo, project := createProject(ctx, t, giteaClient, gwUser01Client, withVisibility(gwapitypes.VisibilityPrivate))
+
+	push(t, config, giteaRepo.CloneURL, giteaToken, "commit", false)
+
+	runNumbers := 5
+
+	var startRunNumber uint64
+	for i := 0; i < runNumbers; i++ {
+		_ = testutil.Wait(30*time.Second, func() (bool, error) {
+			runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, startRunNumber, 0, true)
+			if err != nil {
+				return false, nil
+			}
+
+			if len(runs) == 0 {
+				return false, nil
+			}
+			if runs[0].Phase != rstypes.RunPhaseFinished {
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+		runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, startRunNumber, 0, true)
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+		if len(runs) == 0 {
+			t.Fatalf("expected 1 run got: %d", len(runs))
+		}
+		if runs[0].Phase != rstypes.RunPhaseFinished {
+			t.Fatalf("expected run phase %q, got %q", rstypes.RunPhaseFinished, runs[0].Phase)
+		}
+		if runs[0].Result != rstypes.RunResultSuccess {
+			t.Fatalf("expected run result %q, got %q", rstypes.RunResultSuccess, runs[0].Result)
+		}
+
+		startRunNumber = runs[0].Number - 1
+
+		if i == runNumbers-1 {
+			break
+		}
+
+		if _, err = gwUser01Client.ProjectCreateRun(ctx, project.ID, &gwapitypes.ProjectCreateRunRequest{Branch: "master"}); err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+	}
```

Comment:
Why don't you just call 'push' n times and check that all the runs are finished/successfull instead of 1 time + all of this code?

Reply:
I did the change but it fails with setuperror, I have seen it happen because something is wrong with gitea when the runservice get the conf file.. it can be because of too many calls to gitea during the run creation(?). This is why I did it by a for cicle before. I could add a time sleep after evry ProjectCreateRun if it works?